astropy/modeling/astro_sherpa.py

```diff
@@ -0,0 +1,616 @@
+            elif len(models) == self._data.ndata:
+                self._fitmodel = ConvertedModel(models, tie_list)
+            else:
+                raise Exception("Don't know how to handle multiple models unless there is one foreach dataset")
+        else:
+            if len(models) > 1:
+                self._data.make_simfit(len(models))
+                self._fitmodel = ConvertedModel(models, tie_list)
+            else:
+                self._fitmodel = ConvertedModel(models)
+
+        self._fitter = Fit(self._data.data, self._fitmodel.sherpa_model, self._stat_method, self._opt_method, self._est_method, **kwargs)
+        self.fit_info = self._fitter.fit()
+
+        return self._fitmodel.get_astropy_model()
+
+    def est_errors(self, sigma=None, maxiters=None, numcores=1, methoddict=None, parlist=None):
+        """
+        Use sherpa error estimators based on the last fit.
+
+        Parameters:
+            sigma: float
+                this will be set as the confidance interval for which the errors are found too.
+            maxiters: int
+                the maximum number of iterations the error estimator will run before giving up.
+            methoddict: dict
+                !not quite sure couldn't figure this one out yet!
+            parlist: list
+                a list of parameters to find the confidance interval of if none are provided all free
+                parameters will be estimated.
+        """
+        if self._fitter is None:
+            ValueError("Must complete a valid fit before errors can be calculated")
+        if sigma is not None:
+            self._fitter.estmethod.config['sigma'] = sigma
+        if maxiters is not None:
+            self._fitter.estmethod.config['maxiters'] = maxiters
+        if 'numcores' in self._fitter.estmethod.config:
+            if not numcores == self._fitter.estmethod.config['numcores']:
+                self._fitter.estmethod.config['numcores'] = numcores
+
+        return self._fitter.est_errors(methoddict=methoddict, parlist=parlist)
+
+
+class Dataset(SherpaWrapper):
+
+    """
+    Parameters
+        ----------
+        n_dim: int
+            Used to veirfy required number of dimentions.
+        x : array (or list of arrays)
+            input coordinates
+        y : array (or list of arrays)
+            input coordinates
+        z : array (or list of arrays) (optional)
+            input coordinates
+        xerr : array (or list of arrays) (optional)
+            an array of errors in x
+        yerr : array (or list of arrays) (optional)
+            an array of errors in y
+        zerr : array (or list of arrays) (optional)
+            an array of errors in z
+
+    returns:
+        _data: a sherpa dataset
+    """
+
+    def __init__(self, n_dim, x, y, z=None, xerr=None, yerr=None, zerr=None, bkg=None, bkg_scale=1):
+
+        x = np.array(x)
+        y = np.array(y)
+        if x.ndim == 2 or (x.dtype == np.object or y.dtype == np.object):
```

Comment:
if the array is made from a  uneven lengths then it just sets each array as an object so ndim will be 1 instead of 2